daft/udf.py

```diff
@@ -193,6 +252,8 @@ def override_options(
             if memory_bytes is not _UnsetMarker:
                 new_resource_request = new_resource_request.with_memory_bytes(memory_bytes)
             result = dataclasses.replace(result, resource_request=new_resource_request)
+            if batch_size is not _UnsetMarker:
+                result.batch_size = batch_size
 
```

Comment:
Let's separate the resource requests code and the batch_size code?

This can be outside the `if not all(...)` guard, which I put in place because those resource requests had to modify the same `new_resource_request` object

```python
if batch_size is not _UnsetMarker:
    result.batch_size = batch_size
```